synapseclient/models/mixins/table_operator.py

```diff
@@ -0,0 +1,2836 @@
+            from synapseclient.models import Column, ColumnType, Table
+
+            syn = Synapse()
+            syn.login()
+
+            async def main():
+                table = await Table(
+                    id="syn1234"
+                ).get_async(include_columns=True)
+
+                # Move the column to the beginning of the list
+                table.reorder_column(name="my_column", index=0)
+                table.store_async()
+
+            asyncio.run(main())
+            ```
+        """
+        if not self._last_persistent_instance:
+            raise ValueError(
+                "This method is only supported after interacting with Synapse via a `.get()` or `.store()` operation"
+            )
+
+        column_to_reorder = self.columns.pop(name, None)
+        if index >= len(self.columns):
+            self.columns[name] = column_to_reorder
+            return self
+
+        self.columns = OrderedDict(
+            list(self.columns.items())[:index]
+            + [(name, column_to_reorder)]
+            + list(self.columns.items())[index:]
+        )
+
+    @staticmethod
+    def _convert_columns_to_ordered_dict(
+        columns: Union[
+            List["Column"], OrderedDict[str, "Column"], Dict[str, "Column"], None
+        ]
+    ) -> OrderedDict[str, "Column"]:
+        """Converts the columns attribute to an OrderedDict if it is a list or dict."""
+        if not columns:
+            return OrderedDict()
+
+        if isinstance(columns, list):
+            results = OrderedDict()
+            for column in columns:
+                if column.name in results:
+                    raise ValueError(f"Duplicate column name: {column.name}")
+                results[column.name] = column
+            return results
+        elif isinstance(columns, dict):
+            results = OrderedDict()
+            for key, column in columns.items():
+                if column.name:
+                    if column.name in results:
+                        raise ValueError(f"Duplicate column name: {column.name}")
+                    results[column.name] = column
+                else:
+                    column.name = key
+                    if key in results:
+                        raise ValueError(f"Duplicate column name: {key}")
+                    results[key] = column
+            return results
+        elif isinstance(columns, OrderedDict):
+            results = OrderedDict()
+            for key, column in columns.items():
+                if column.name:
+                    if column.name in results:
+                        raise ValueError(f"Duplicate column name: {column.name}")
+                    results[column.name] = column
+                else:
+                    column.name = key
+                    if key in results:
+                        raise ValueError(f"Duplicate column name: {key}")
+                    results[key] = column
+            return results
```

Comment:
Could these be combined into one block that checks for either type of dictionary?

Reply:
Changes made in this commit: https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1168